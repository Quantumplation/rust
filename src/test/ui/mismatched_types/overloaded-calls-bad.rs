#![feature(fn_traits, unboxed_closures)]

use std::ops::FnMut;

struct S {
    x: isize,
    y: isize,
}

impl FnMut<(isize,)> for S {
    extern "rust-call" fn call_mut(&mut self, (z,): (isize,)) -> isize {
        self.x * self.y * z
    }
}

impl FnOnce<(isize,)> for S {
    type Output = isize;
    extern "rust-call" fn call_once(mut self, (z,): (isize,)) -> isize {
        self.call_mut((z,))
    }
}

fn main() {
    let mut s = S {
        x: 3,
        y: 3,
    };
    let ans = s("what");    //~ ERROR arguments to this function are incorrect
    let ans = s();
    //~^ ERROR arguments to this function are incorrect
    let ans = s("burma", "shave");
    //~^ ERROR arguments to this function are incorrect
}
