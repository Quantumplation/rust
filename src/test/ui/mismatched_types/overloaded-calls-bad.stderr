error[E0308]: arguments to this function are incorrect
  --> $DIR/overloaded-calls-bad.rs:28:15
   |
LL |     let ans = s("what");
   |               ^^------^
   |                 |
   |                 expected `isize`, found `&'static str`
   |
help: provide an argument of the correct type
   |
LL |     let ans = ({isize});
   |               ^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/overloaded-calls-bad.rs:29:15
   |
LL |     let ans = s();
   |               ^^^ an argument of type isize is missing
   |
help: provide the argument
   |
LL |     let ans = ({isize});
   |               ^^^^^^^^^

error[E0308]: arguments to this function are incorrect
  --> $DIR/overloaded-calls-bad.rs:31:15
   |
LL |     let ans = s("burma", "shave");
   |               ^^-------^^^^^^^^^^
   |                 |
   |                 expected `isize`, found `&'static str`
   |                 argument of type isize unexpected
   |
help: did you mean
   |
LL |     let ans = ({isize});
   |               ^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
